lock/src/main/java/org/infinispan/lock/impl/lock/ClusteredLockImpl.java

```diff
@@ -264,8 +365,13 @@ private void lock(RequestHolder<Void> requestHolder) {
          return;
 
       pendingRequests.offer(requestHolder);
```

Comment:
I don't see where pendingRequests is removed if the lock is acquire below in the eval method?

Reply:
it is removed on the listener, done requests are removed and skipped while seeking for the next requests to be treated. Do you think it is worth to remove it from the pending requests too to avoid memory issues ?

Reply:
I was more wondering so we don't try to retry a lock accidentally.

Reply:
well, I think today the risk is doing more than one rpc call for the same request, indeed.
The listener notification can arrive before the rpc call finishes, so we can retry the same request. I need to check in the listener that the request that acquired the lock is not retried